src/lib/AllocationRequestCbor.sol

```diff
@@ -44,4 +98,17 @@ library AllocationRequestCbor {
 
         return CBOR.data(buf);
     }
+
+    function _calculateDataArrayPrefix(uint dataSize) internal pure returns (uint) {
```

Comment:
Use filecoin library for that: https://github.com/filecoin-project/filecoin-solidity/blob/master/contracts/v0.8/utils/Misc.sol#L61